targets/simple_switch/primitives.cpp

```diff
@@ -473,6 +475,66 @@ class truncate_ : public ActionPrimitive<const Data &> {
 
 REGISTER_PRIMITIVE_W_NAME("truncate", truncate_);
 
+class log_msg : public ActionPrimitive<const std::string &, std::vector<Data>> {
```

Comment:
please use a const reference for the vector, there is no need to pay for a copy here I think

Reply:
I think we need to pay for a copy here. We make new local variable of vector<Data> in "ActionParam::to" function, and we cant use reference to a local variable. If I misunderstood something here please let me know.